run.py

```diff
@@ -228,6 +228,22 @@ def before_first_request():
     os.makedirs(os.path.dirname(os.path.join(app.config['LOGS_DIR'])), exist_ok=True)
     os.makedirs(os.path.dirname(os.path.join(app.config['OUTPUT_DIR'])), exist_ok=True)
 
+@app.errorhandler(404)
+def page_not_found_handler(e):
+    """
+    Handles the 404 error and displays a template.
+    """
+    layout = LayoutHelper(logged_in=False)
```

Comment:
A user could be logged in and still encounter a 404 ;)

Reply:
Good point, will fix that